describegroups.go

```diff
@@ -0,0 +1,176 @@
+	return sizeofString(t.MemberID) +
+		sizeofString(t.ClientID) +
+		sizeofString(t.ClientHost) +
+		sizeofBytes(t.MemberMetadata) +
+		sizeofBytes(t.MemberAssignments)
+}
+
+func (t describeGroupsResponseMemberV1) writeTo(w *bufio.Writer) {
+	writeString(w, t.MemberID)
+	writeString(w, t.ClientID)
+	writeString(w, t.ClientHost)
+	writeBytes(w, t.MemberMetadata)
+	writeBytes(w, t.MemberAssignments)
+}
+
+func (t *describeGroupsResponseMemberV1) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if remain, err = readString(r, size, &t.MemberID); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.ClientID); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.ClientHost); err != nil {
+		return
+	}
+	if remain, err = readBytes(r, remain, &t.MemberMetadata); err != nil {
+		return
+	}
+	if remain, err = readBytes(r, remain, &t.MemberAssignments); err != nil {
+		return
+	}
+	return
+}
+
+type describeGroupsResponseGroupV1 struct {
+	// ErrorCode holds response error code
+	ErrorCode int16
+
+	// GroupID holds the unique group identifier
+	GroupID string
+
+	// State holds current state of the group (one of: Dead, Stable, AwaitingSync,
+	// PreparingRebalance, or empty if there is no active group)
+	State string
+
+	// ProtocolType holds the current group protocol type (will be empty if there is
+	// no active group)
+	ProtocolType string
+
+	// Protocol holds the current group protocol (only provided if the group is Stable)
+	Protocol string
+
+	// Members contains the current group members (only provided if the group is not Dead)
+	Members []describeGroupsResponseMemberV1
+}
+
+func (t describeGroupsResponseGroupV1) size() int32 {
+	return sizeofInt16(t.ErrorCode) +
+		sizeofString(t.GroupID) +
+		sizeofString(t.State) +
+		sizeofString(t.ProtocolType) +
+		sizeofString(t.Protocol) +
+		sizeofArray(len(t.Members), func(i int) int32 { return t.Members[i].size() })
+}
+
+func (t describeGroupsResponseGroupV1) writeTo(w *bufio.Writer) {
+	writeInt16(w, t.ErrorCode)
+	writeString(w, t.GroupID)
+	writeString(w, t.State)
+	writeString(w, t.ProtocolType)
+	writeString(w, t.Protocol)
+	writeArray(w, len(t.Members), func(i int) { t.Members[i].writeTo(w) })
+}
+
+func (t *describeGroupsResponseGroupV1) readFrom(r *bufio.Reader, size int) (remain int, err error) {
+	if remain, err = readInt16(r, size, &t.ErrorCode); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.GroupID); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.State); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.ProtocolType); err != nil {
+		return
+	}
+	if remain, err = readString(r, remain, &t.Protocol); err != nil {
+		return
+	}
+
+	remain, err = readArrayWith(r, remain, func(r *bufio.Reader, size int) (remain int, err error) {
+		item := describeGroupsResponseMemberV1{}
+		remain, err = (&item).readFrom(r, size)
+		t.Members = append(t.Members, item)
```

Comment:
Should we check for `err == nil` before appending the value here? Otherwise the last member appended to the list is very likely meaningless.

Reply:
Whoops.  May have accidentally deleted my comment.

I can change this if you would like me to, but I'd argue that when err != nil the entire contents of describeGroupsResponseMemberV1 are meaningless with the possible exception of ErrorCode and ErrorMessage which aren't impacted by this pattern.

My guess is that you'd probably question code that tried to make use of a value associated with err != nil.  For example, consider ```err := json.Unmarshal(data, &content)```.  If err != nil, I would consider the entire value of ```content``` suspect.

Happy to change if you feel strongly about this.

Reply:
Honestly I don’t feel too strong about either ways, if we don’t leak the value at a higher level it’s kind of OK, but there are places where you’ve checked for the error before appending and places where you don’t. I care more about consistency so it doesn’t mislead the reader into questioning why errors are sometimes checked and sometimes not. I guess my advise would be to pick one programming style and apply it everywhere there’s a similar code pattern.

Reply:
It's easy enough to change.